libs/model/src/middleware/guards.ts

```diff
@@ -106,3 +107,53 @@ export function mustBeAuthorizedComment(actor: Actor, auth?: AuthContext) {
     comment_id: number;
   };
 }
+
+/**
+ * Guards for starting and ending dates to be in a valid date range
+ * @param start_date start date
+ * @param end_date end date
+ * @param startDaysInTheFuture number of days in the future that the start date must be
+ * @param minDaysInRange
+ * @returns { start, end } start and end dates as moment objects
+ */
+export function mustBeValidDateRange(
+  start_date: Date,
+  end_date: Date,
+  startDaysInTheFuture = 1,
+  minDaysInRange = 1,
+) {
+  const today = moment(new Date());
+  const start = moment(start_date);
+  const end = moment(end_date);
+
+  // check if start date is in the future by more than startDaysInTheFuture
+  const firstValidStartDate = today.add(startDaysInTheFuture, 'days');
+  if (start.isBefore(firstValidStartDate))
+    throw new InvalidState(
+      `Start date ${start.format('YYYY-MM-DD')} must be at least ${startDaysInTheFuture} days in the future`,
+      { start_date, end_date },
+    );
+
+  // check that end date is after start date by more than minDaysInRange
+  const rangeInDays = end.diff(start, 'days');
+  if (rangeInDays < minDaysInRange)
+    throw new InvalidState(
+      `Start ${start.format('YYYY-MM-DD')} and End ${end.format('YYYY-MM-DD')} dates range must be at least ${minDaysInRange} days apart, but are ${rangeInDays} days apart`,
+      { start_date, end_date },
+    );
+
+  return { start, end };
+}
```

Comment:
Isolated unit test would be nice e.g. to check for off-by-one error, but up to you

Reply:
we can if coverage is not 100%, but I think we'll cover the most important parts with lifecycle tests